### PR TITLE
Moves spare to safe located in bridge, acting captain gets codes to it

### LINKED ISSUE #3832: Bounty: Give Acting Captain Access to Captain Quarters (35$)
If the new rule set is going to require the Acting Captain get the spare ID and nuke disc before the riders, the code should reflect those rules and give the acting captain access to do so. Right now acting captain needs to be excellent at breaking and entering, that has little to do with managing the station and running their chosen department. Let's make the code base reflect the rules.

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Spare has been moved from caps locker to safe located in bridge.
Paper with codes to that safe is given to captain, if there is one roundstart.
If there is no captain and if `SPARE_ENFORCE_COC` config entry is uncommented it will give the codes to the head highest in CoC. If the `SPARE_ENFORCE_COC` is not enabled it will give the codes to every head present on roundstart.

some code and general idea of keeping spare in safe taken from: https://github.com/tgstation/tgstation/pull/56910

Current chain of command:
1. Captain
2. HoP
3. RD
4. CE
5. CMO
6. HoS

SPARE_ENFORCE_COC is enabled on sage and not enabled on golden.

closes: #3832
alternative to and closes: #3855

## Why It's Good For The Game

No more breaking into the locker with fireaxe roundstart.

## Changelog
:cl:
add: Spare has been moved to safe located in bridge.
add: Captain gets code to that safe.
config: New config entry SPARE_ENFORCE_COC.
config: If there is no captain and SPARE_ENFORCE_COC is enabled code will be given to head highest in CoC.
config: If there is no captain and SPARE_ENFORCE_COC is not enabled code will be given to all of roundstart heads.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
This should follow the chain of command.

**Comment by leestriter** [comment]
What does everyone think would work better, this or the vault method? Bloons made a pretty good case for the vault method as it doesn't force captainship. But I'm for whatever the community/coders think is best for the game. Also, thank you archanial. I'll go close the issue and get you your bounty.

**Comment by leestriter** [comment]
Never used bountysource before. How do I get archanial his bounty?

**Comment by leestriter** [comment]
> This should follow the chain of command.

Also this.

**Comment by leestriter** [comment]
Sorry to be difficult, the more I think about it, the more I prefer the safe option for enabling any head to secure the disk and ID. It lets any head handle it, and doesnt force captainship on people.

**Comment by Archanial** [comment]
I've talked with leestriter in dms and we've decided to go with safe and make it follow CoC.

**Comment by Archanial** [comment]
Changes have been made, this is ready for review.

**Comment by St0rmC4st3r** [comment]
> Never used bountysource before. How do I get archanial his bounty?

I advise you to avoid bountysource. It has multiple problems with using it.

**Comment by utilbird** [comment]
clash of clans

File: code/controllers/subsystem/job.dm
Comment (utilbird) [inline code/controllers/subsystem/job.dm]: Probably don't need defines for this as they're only used once, you can just put the strings in the list

File: code/game/objects/items/storage/secure.dm
Comment (utilbird) [inline code/game/objects/items/storage/secure.dm]:
```suggestion
/obj/item/storage/secure/safe/caps_spare/Initialize()
	. = ..()
	l_code = SSjob.spare_id_safe_code
	l_set = TRUE
	SEND_SIGNAL(src, COMSIG_TRY_STORAGE_SET_LOCKSTATE, TRUE)
```

File: code/game/objects/items/storage/secure.dm
Comment (utilbird) [inline code/game/objects/items/storage/secure.dm:208]: if the safe is going to be this strong, there should probably be more alternative methods for breaking in imo. Like thermite

File: code/game/objects/items/storage/secure.dm
Comment (utilbird) [inline code/game/objects/items/storage/secure.dm]:
```suggestion
			if(open)
```

File: code/controllers/subsystem/job.dm
Comment (Archanial) [reply code/controllers/subsystem/job.dm]: It's undefined at the end.
Defines are only used by compiler, there is no reason to create list, you are going to use once and not modify.

**Comment by FlamePrince** [comment]
I've been playing a lot of Golden as CE, and this has resulted numerous times in my being acting captain. I have to abandon my duties as CE to the engine to rush Captain's locker to try and get the spare before some tider does. This has caused some scuffles and I think securing it behind something that can't be destroyed easily by the readily available fire axe is a great idea that'll help cut down on some of the greytiding that leads to everyone having AA.

That being said, this will not be a community favourite. Once this is merged and greytiders can no longer lol themselves into AA, there could be a significant hit to the playerbase of Golden.

File: code/game/objects/items/storage/secure.dm
Comment (Archanial) [reply code/game/objects/items/storage/secure.dm:208]: You can still break with thermite or explosion. The preferred method would be to kill acting captain and get spare that way. Numbers look strong we don't want to have people rushing fireaxe to break in.

**Review by utilbird — APPROVED** [review]

**Comment by utilbird** [comment]
Looks fine codewise. Waiting on input from other maintainers, else I'll put it up for vote

**Comment by leestriter** [comment]
> I've been playing a lot of Golden as CE, and this has resulted numerous times in my being acting captain. I have to abandon my duties as CE to the engine to rush Captain's locker to try and get the spare before some tider does. This has caused some scuffles and I think securing it behind something that can't be destroyed easily by the readily available fire axe is a great idea that'll help cut down on some of the greytiding that leads to everyone having AA.
> 
> That being said, this will not be a community favourite. Once this is merged and greytiders can no longer lol themselves into AA, there could be a significant hit to the playerbase of Golden.

I commissioned this and primarily play golden. A tider, with only a little more effort, will still be able to get AA, but now someone playing a head should be able to get before a tider if both are around headstart, as it should be. The problem was in absence of a captain, whatever head is there is responsible in the rules for securing the spare from tiders, who are, in the golden rules, allowed to take the spare but become valid. AA currently takes under 5 minutes to steal for a skilled tider. This PR was funded specifically with improving gameplay on golden in mind and enabling a head on golden to follow the rules to stay out of unneeded trouble with admemes, and make playing a head on golden when there isn't a captain more feasible.

If it doesn't win the vote, we will have to figure out some system so I can pay archanial, as I think bountysource wants something to be merged before it pays out? Anyone with a better understanding of it let me know. I intend to pay if it is merged or not, so maybe even if it loses there's some way to merge and remove so I can pay archanial? Hopefully it wins the vote, I think this is a positive change.

**Comment by FlamePrince** [comment]
> make playing a head on golden when there isn't a captain more feasible.

/thread

also, i'd like to point out that this PR may pave way, whether good or bad, to moving the nuke disk into the same safe and removing the niche requirement for it to be "secured". while smart (acting) Captains will go a farther length to secure it, i think it makes more sense logistically that it is "secured" at roundstart then laying out on a table to potentially be forgotten about and then someone gets bwoinked when it goes missing

**Comment by Vexylius** [comment]
Putting HoS at the bottom of this list might be something to talk about. I know that HoS as an acting cap can auth executions, and that's there to prevent power trips and stuff, but like usually when there is no HoP present, captainship is handed over to HoS first, and there are admins that can handle bad acting-cap HoSes.

**Comment by Glcbitj** [comment]
> Putting HoS at the bottom of this list might be something to talk about. I know that HoS as an acting cap can auth executions, and that's there to prevent power trips and stuff, but like usually when there is no HoP present, captainship is handed over to HoS first, and there are admins that can handle bad acting-cap HoSes.

The HoS is probably the one out of the heads expected to be the more "robust", so there's some sense in putting him there with a higher priority. But i think the main reason of that is because HoS is already expected ~~validhunt~~ handle security matters, which can demand a bunch of his time and focus, not to mention that he'll need to put himself in dangerous situations, meanwhile CE is free after the SM is set up if there's no engineering emergencies, RD is free if there is one dude to press R&D console buttons, CMO is free if medbay is not a looming pile of corpses (so maybe not so much), and HoP only needs to open the line sometimes and maybe handle cargo if there's no QM.

Buuut, HoP and HoS do are the only heads besides Cap that do start with Mindshields, so there's that too if you take cult stealthy conversions in account i guess.

**Comment by PowerfulBacon** [comment]
> SPARE_ENFORCE_COC is enabled on sage and not enabled on golden.

Why not?

**Review by francinum — APPROVED** [review]

**Comment by FlamePrince** [comment]
Just played a round as Captain, didn't receive the piece of paper with the safe code.

**Pull request merged**
